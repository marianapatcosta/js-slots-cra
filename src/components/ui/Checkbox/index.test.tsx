import { render, screen, fireEvent } from '@testing-library/react';
import { Checkbox } from './index';

describe('Checkbox', () => {
  const defaultProps = {
    label: 'Checkbox label',
    checked: false,
    onChange: () => null,
  };
  it('renders without errors and matches snapshot', () => {
    const {
      container: { firstChild },
    } = render(<Checkbox {...defaultProps} />);

    expect(firstChild).toMatchSnapshot();
  });

  it('should display label', () => {
    render(<Checkbox {...defaultProps} />);

    screen.getByText('Checkbox label');
  });

  it('should handle change events', () => {
    const onChangeMock = jest.fn();
    render(<Checkbox {...defaultProps} onChange={onChangeMock} />);
    const checkbox = screen.getByLabelText(/Checkbox Label/i) as HTMLInputElement;
    fireEvent.click(checkbox);

    expect(onChangeMock).toBeCalled();
  });
});
