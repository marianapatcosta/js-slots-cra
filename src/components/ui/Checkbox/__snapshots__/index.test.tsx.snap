// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders without errors and matches snapshot 1`] = `
<label
  class="checkbox  "
>
  <input
    type="checkbox"
  />
  <div
    class="checkbox__toggle"
  />
  Checkbox label
</label>
`;
